### ISSUE TITLE
datalad-archives download failure (works with git-annex get)

### ISSUE THREAD

**Issue by mih** [datalad/datalad#4779]
I want a file that is in a ZIP archive that is available on a remote:

```
% git annex whereis ses-2/anat/sub-1000201_ses-2_FLAIR.json               
whereis ses-2/anat/sub-1000201_ses-2_FLAIR.json (1 copy) 
        c04eb54b-4b4e-5755-8436-866b043170fa -- [datalad-archives]

  datalad-archives: dl+archive:MD5E-s33053223--e825ec7be8b391ddd463e51d6b33aee2.zip#path=T2_FLAIR/T2_FLAIR.json&size=313
ok
```

I cannot `datalad get` it.

```
% datalad get ses-2/anat/sub-1000201_ses-2_FLAIR.json 
[INFO   ] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO   ] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO   ] To obtain some keys we need to fetch an archive of size 33.1 MB 
[WARNING] Running get resulted in stderr output: [INFO] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO] PROGRESS-JSON: {"command":"get","wanted":[{"here":false,"uuid":"00000000-0000-0000-0000-000000000001","description":"web"}],"note":"transfer already in progress, or unable to take transfer lock\nUnable to access these remotes: web\nTry making some of these repositories available:\n\t00000000-0000-0000-0000-000000000001 -- web\n\n(Note that these git remotes have annex-ignore set: origin)","skipped":[],"success":false,"key":"MD5E-s33053223--e825ec7be8b391ddd463e51d6b33aee2.zip","file":null} 
[INFO] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO] PROGRESS-JSON: {"command":"get","wanted":[{"here":false,"uuid":"00000000-0000-0000-0000-000000000001","description":"web"}],"note":"transfer already in progress, or unable to take transfer lock\nUnable to access these remotes: web\nTry making some of these repositories available:\n\t00000000-0000-0000-0000-000000000001 -- web\n\n(Note that thes... 
...
[ERROR  ] from datalad-archives...; from datalad-archives...; from datalad-archives...; Unable to access these remotes: datalad-archives; Try making some of these repositories available:;     c04eb54b-4b4e-5755-8436-866b043170fa -- [datalad-archives]; ; (Note that these git remotes have annex-ignore set: origin) [get(/data/project/ukb_vbm/65dfd6d4-d1b0-11ea-9471-b4969146869c/ses-2/anat/sub-1000201_ses-2_FLAIR.json)] 
get(error): /data/project/ukb_vbm/65dfd6d4-d1b0-11ea-9471-b4969146869c/ses-2/anat/sub-1000201_ses-2_FLAIR.json (file) [from datalad-archives...; from datalad-archives...; from datalad-archives...; Unable to access these remotes: datalad-archives; Try making some of these repositories available:;        c04eb54b-4b4e-5755-8436-866b043170fa -- [datalad-archives]; ; (Note that these git remotes have annex-ignore set: origin)]
```

Multiple attempts do not change the result. However, I can manually obtain the key of the archive, and then it works:

```
% git annex get --key MD5E-s33053223--e825ec7be8b391ddd463e51d6b33aee2.zip
get MD5E-s33053223--e825ec7be8b391ddd463e51d6b33aee2.zip (from web...) 
(checksum...) ok                  
(recording state in git...)
% datalad get ses-2/anat/sub-1000201_ses-2_FLAIR.json                     
get(ok): /data/project/ukb_vbm/65dfd6d4-d1b0-11ea-9471-b4969146869c/ses-2/anat/sub-1000201_ses-2_FLAIR.json (file) [from datalad-archives...]
```

**Comment by mih** [comment]
Tried exact same on a different machine, with the exact same dataset, cloned from the exact same source -- works!

Both machines have 0.13.1 (failing one installed from pip, working one from Debian package). Other major difference is git annex:

- broken: 7.20190819+git2-g908476a9b-1~ndall+1 (standalone)
- working: 8.20200330 (standard package)

**Comment by yarikoptic** [comment]
is that dataset something you still have access to try on, or should we just let this issue RiP after these years ?

actually looking at ukb_vbm  suggests that authentication was somehow involved, and I remember various fixes since those times on preventing parallel operation  and I see in the logs duplicate log messages:

```
% datalad get ses-2/anat/sub-1000201_ses-2_FLAIR.json 
[INFO   ] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO   ] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO   ] To obtain some keys we need to fetch an archive of size 33.1 MB 
[WARNING] Running get resulted in stderr output: [INFO] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO] PROGRESS-JSON: {"command":"get","wanted":[{"here":false,"uuid":"00000000-0000-0000-0000-000000000001","description":"web"}],"note":"transfer already in progress, or unable to take transfer lock\nUnable to access these remotes: web\nTry making some of these repositories available:\n\t00000000-0000-0000-0000-000000000001 -- web\n\n(Note that these git remotes have annex-ignore set: origin)","skipped":[],"success":false,"key":"MD5E-s33053223--e825ec7be8b391ddd463e51d6b33aee2.zip","file":null} 
[INFO] To obtain some keys we need to fetch an archive of size 33.1 MB 
[INFO] PROGRESS-JSON: {"command":"get","wanted":[{"here":false,"uuid":"00000000-0000-0000-0000-000000000001","description":"web"}],"note":"transfer already in progress, or unable to take transfer lock\nUnable to access these remotes: web\nTry making some of these repositories available:\n\t00000000-0000-0000-0000-000000000001 -- web\n\n(Note that thes... 
...
[ERROR  ] from datalad-archives...; from datalad-archives...; from datalad-archives...; Unable to access these remotes: datalad-archives; 

```

and then it was not reproducible on some other instance ... so I think we would just let this issue RiP but feel welcome to reopen if remains pertinent.
